### DISCUSSION TITLE
pdn with different layers due to obstruction

### DISCUSSION THREAD

**Discussion by titan73** [The-OpenROAD-Project/OpenROAD#5963] [Q&A] [answered]
In the picture from Innovus, the are M3 horizontal straps (yellow) and M4 vertical straps (green).
Due to M3 obstructions (yellow), the vertical straps switch layer to M2 (red). This is achieve using the -area argument au add_stripes.

![image](https://github.com/user-attachments/assets/63cfd372-06f9-4811-9958-5d9194492c08)

Is there a way to do a similar thing in OR?

**Comment by maliberty** [comment]
@gadfort

**Reply by gadfort** [reply, marked as answer]
@titan73 the short answer is no, pdn does not have a mechanism to this. Adding a strap on M2 would also require avoiding placing standard cells under the strap as there may not be a way to route to them. PDN will honor obstructions if added before `pdngen` runs, but in this case that would leave the bottom of your circuit disconnected.

**Reply by titan73** [reply]
Thanks for the answer. I'll enter an enhancement request for that.
